// cordova-res/dist/resources.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const path_1 = tslib_1.__importDefault(require("path"));
const util_1 = tslib_1.__importDefault(require("util"));
const error_1 = require("./error");
const platform_1 = require("./platform");
exports.RESOURCE_TYPES = [
    "adaptive-icon" /* ADAPTIVE_ICON */,
    "icon" /* ICON */,
    "splash" /* SPLASH */,
];
exports.RESOURCE_FORMATS = ["jpeg" /* JPEG */, "png" /* PNG */];
exports.RESOURCE_RASTER_FORMATS = [
    "jpeg" /* JPEG */,
    "png" /* PNG */,
];
function isResourceFormat(format) {
    return exports.RESOURCE_FORMATS.includes(format);
}
exports.isResourceFormat = isResourceFormat;
function isRasterResourceFormat(format) {
    return exports.RESOURCE_RASTER_FORMATS.includes(format);
}
exports.isRasterResourceFormat = isRasterResourceFormat;
async function validateRasterResource(platform, type, source, metadata, schema) {
    const { format, width, height } = metadata;
    const { width: requiredWidth, height: requiredHeight } = schema;
    if (!format || !isRasterResourceFormat(format)) {
        throw new error_1.ValidationError(`The format for source image of type "${type}" must be one of: (${exports.RESOURCE_RASTER_FORMATS.join(', ')}) (image format is "${format}").`, {
            source,
            type,
            code: "BAD_IMAGE_FORMAT" /* BAD_IMAGE_FORMAT */,
            format,
            requiredFormats: exports.RESOURCE_RASTER_FORMATS,
        });
    }
    if (!width || !height || width < requiredWidth || height < requiredHeight) {
        throw new error_1.ValidationError(`The dimensions for source image of type "${type}" do not meet minimum size requirements: ${requiredWidth}x${requiredHeight} (image is ${width}x${height}).`, {
            source,
            type,
            code: "BAD_IMAGE_SIZE" /* BAD_IMAGE_SIZE */,
            width,
            height,
            requiredWidth,
            requiredHeight,
        });
    }
}
exports.validateRasterResource = validateRasterResource;
exports.COLOR_REGEX = /^\#[A-F0-9]{6}$/;
function getRasterResourceSchema(platform, type) {
    switch (type) {
        case "adaptive-icon" /* ADAPTIVE_ICON */:
            return { width: 432, height: 432 };
        case "icon" /* ICON */:
            return { width: 1024, height: 1024 };
        case "splash" /* SPLASH */:
            return { width: 2732, height: 2732 };
    }
}
exports.getRasterResourceSchema = getRasterResourceSchema;
async function validateResource(platform, type, source, pipeline, errstream) {
    const metadata = await pipeline.metadata();
    const schema = getRasterResourceSchema(platform, type);
    await validateRasterResource(platform, type, source, metadata, schema);
    if (platform === "ios" /* IOS */ && type === "icon" /* ICON */) {
        if (metadata.hasAlpha) {
            // @see https://github.com/ionic-team/cordova-res/issues/94
            errstream === null || errstream === void 0 ? void 0 : errstream.write(util_1.default.format('WARN:\tSource icon %s contains alpha channel, generated icons for %s will not.\n\n' +
                '\tApple recommends avoiding transparency. See the App Icon Human Interface Guidelines[1] for details. Any transparency in your icon will be filled in with white.\n\n' +
                '\t[1]: https://developer.apple.com/design/human-interface-guidelines/ios/icons-and-images/app-icon/\n', source, platform_1.prettyPlatform(platform)) + '\n');
        }
    }
    return metadata;
}
exports.validateResource = validateResource;
function validateResourceTypes(types) {
    const result = [];
    for (const type of types) {
        if (!isSupportedResourceType(type)) {
            throw new error_1.BadInputError(`Unsupported resource type: ${type}`);
        }
        result.push(type);
    }
    return result;
}
exports.validateResourceTypes = validateResourceTypes;
function isSupportedResourceType(type) {
    return exports.RESOURCE_TYPES.includes(type);
}
exports.isSupportedResourceType = isSupportedResourceType;
function getSimpleResources(platform, type) {
    switch (platform) {
        case "android" /* ANDROID */:
            switch (type) {
                case "icon" /* ICON */:
                    return exports.ANDROID_ICON_RESOURCES;
                case "splash" /* SPLASH */:
                    return exports.ANDROID_SPLASH_RESOURCES;
            }
            break;
        case "ios" /* IOS */:
            switch (type) {
                case "icon" /* ICON */:
                    return exports.IOS_ICON_RESOURCES;
                case "splash" /* SPLASH */:
                    return exports.IOS_SPLASH_RESOURCES;
            }
            break;
        case "windows" /* WINDOWS */:
            switch (type) {
                case "icon" /* ICON */:
                    return exports.WINDOWS_ICON_RESOURCES;
                case "splash" /* SPLASH */:
                    return exports.WINDOWS_SPLASH_RESOURCES;
            }
            break;
    }
    throw new error_1.BadInputError(`Unsupported platform/resource type combination: ${platform}/${type}`);
}
exports.getSimpleResources = getSimpleResources;
function generateScaledWindowsResourceSrc(src, factor) {
    const { dir, name, ext } = path_1.default.parse(src);
    return path_1.default.posix.join(dir, `${name}.scale-${factor * 100}${ext}`);
}
exports.generateScaledWindowsResourceSrc = generateScaledWindowsResourceSrc;
function generateScaledWindowsResource(resource, factor) {
    return {
        ...resource,
        src: generateScaledWindowsResourceSrc(resource.src, factor),
        format: "png" /* PNG */,
        target: undefined,
        width: Math.round(resource.width * factor),
        height: Math.round(resource.height * factor),
    };
}
exports.generateScaledWindowsResource = generateScaledWindowsResource;
function generateScaledWindowsResources(resource, factors) {
    return factors.map(factor => generateScaledWindowsResource(resource, factor));
}
exports.generateScaledWindowsResources = generateScaledWindowsResources;
/**
 * App Icon: App list in start menu, task bar, task manager
 */
const WINDOWS_SQUARE_44_X_44_ICON = {
    platform: "windows" /* WINDOWS */,
    type: "icon" /* ICON */,
    src: 'windows/icon/Square44x44Logo.png',
    format: "none" /* NONE */,
    width: 44,
    height: 44,
    target: "Square44x44Logo" /* SQUARE_44_X_44_LOGO */,
};
/**
 * Small tile: Start menu
 */
const WINDOWS_SQUARE_71_X_71_ICON = {
    platform: "windows" /* WINDOWS */,
    type: "icon" /* ICON */,
    src: 'windows/icon/SmallTile.png',
    format: "none" /* NONE */,
    width: 71,
    height: 71,
    target: "Square71x71Logo" /* SQUARE_71_X_71_LOGO */,
};
/**
 * Medium Tile: For Start menu, Microsoft Store listing
 */
const WINDOWS_SQUARE_150_X_150_ICON = {
    platform: "windows" /* WINDOWS */,
    type: "icon" /* ICON */,
    src: 'windows/icon/Square150x150Logo.png',
    format: "none" /* NONE */,
    width: 150,
    height: 150,
    target: "Square150x150Logo" /* SQUARE_150_X_150_LOGO */,
};
/**
 * Large Tile: Start Menu
 */
const WINDOWS_SQUARE_310_X_310_ICON = {
    platform: "windows" /* WINDOWS */,
    type: "icon" /* ICON */,
    src: 'windows/icon/Square310x310Logo.png',
    format: "none" /* NONE */,
    width: 310,
    height: 310,
    target: "Square310x310Logo" /* SQUARE_310_X_310_LOGO */,
};
/**
 * Wide Tile: Start Menu
 */
const WINDOWS_WIDE_310_X_150_LOGO = {
    platform: "windows" /* WINDOWS */,
    type: "icon" /* ICON */,
    src: 'windows/icon/Wide310x150Logo.png',
    format: "none" /* NONE */,
    width: 310,
    height: 150,
    target: "Wide310x150Logo" /* WIDE_310_X_150_LOGO */,
};
/**
 * Store Logo: App installer, Partner Center, the "Report an app" option in the Store, the "Write a review" option in the Store
 */
const WINDOWS_STORE_LOGO = {
    platform: "windows" /* WINDOWS */,
    type: "icon" /* ICON */,
    src: 'windows/icon/StoreLogo.png',
    format: "none" /* NONE */,
    width: 50,
    height: 50,
    target: "StoreLogo" /* STORE_LOGO */,
};
/**
 * @see https://cordova.apache.org/docs/en/latest/config_ref/images.html#windows
 * @see https://docs.microsoft.com/en-us/windows/uwp/design/style/app-icons-and-logos
 * @see https://docs.microsoft.com/en-us/windows/uwp/design/style/app-icons-and-logos#icon-types-locations-and-scale-factors
 */
exports.WINDOWS_ICON_RESOURCES = [
    WINDOWS_SQUARE_44_X_44_ICON,
    ...generateScaledWindowsResources(WINDOWS_SQUARE_44_X_44_ICON, [
        1,
        1.25,
        1.4,
        1.5,
        2,
        2.4,
        4,
    ]),
    WINDOWS_SQUARE_71_X_71_ICON,
    ...generateScaledWindowsResources(WINDOWS_SQUARE_71_X_71_ICON, [
        1,
        1.25,
        1.4,
        1.5,
        2,
        2.4,
        4,
    ]),
    WINDOWS_SQUARE_150_X_150_ICON,
    ...generateScaledWindowsResources(WINDOWS_SQUARE_150_X_150_ICON, [
        1,
        1.25,
        1.4,
        1.5,
        2,
        2.4,
        4,
    ]),
    WINDOWS_SQUARE_310_X_310_ICON,
    ...generateScaledWindowsResources(WINDOWS_SQUARE_310_X_310_ICON, [
        1,
        1.25,
        1.4,
        1.5,
        1.8,
        2,
        4,
    ]),
    WINDOWS_WIDE_310_X_150_LOGO,
    ...generateScaledWindowsResources(WINDOWS_WIDE_310_X_150_LOGO, [
        0.8,
        1,
        1.25,
        1.4,
        1.5,
        1.8,
        2,
        2.4,
        4,
    ]),
    WINDOWS_STORE_LOGO,
    ...generateScaledWindowsResources(WINDOWS_STORE_LOGO, [
        1,
        1.25,
        1.4,
        1.5,
        1.8,
        2,
        2.4,
        4,
    ]),
];
const WINDOWS_SPLASH_SCREEN = {
    platform: "windows" /* WINDOWS */,
    type: "splash" /* SPLASH */,
    src: 'windows/splash/Splash.png',
    format: "none" /* NONE */,
    width: 620,
    height: 300,
    orientation: "landscape" /* LANDSCAPE */,
    target: "SplashScreen" /* SPLASH_SCREEN */,
};
/**
 * @see https://msdn.microsoft.com/en-us/windows/desktop/hh465338
 * @see https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-splashscreen/index.html#windows-specific-information
 */
exports.WINDOWS_SPLASH_RESOURCES = [
    WINDOWS_SPLASH_SCREEN,
    ...generateScaledWindowsResources(WINDOWS_SPLASH_SCREEN, [
        1,
        1.25,
        1.5,
        2,
        4,
    ]),
];
const ANDROID_LDPI_ADAPTIVE_ICON = {
    platform: "android" /* ANDROID */,
    type: "adaptive-icon" /* ADAPTIVE_ICON */,
    foreground: 'android/icon/ldpi-foreground.png',
    background: 'android/icon/ldpi-background.png',
    format: "png" /* PNG */,
    width: 81,
    height: 81,
    density: "ldpi" /* LDPI */,
};
const ANDROID_MDPI_ADAPTIVE_ICON = {
    platform: "android" /* ANDROID */,
    type: "adaptive-icon" /* ADAPTIVE_ICON */,
    foreground: 'android/icon/mdpi-foreground.png',
    background: 'android/icon/mdpi-background.png',
    format: "png" /* PNG */,
    width: 108,
    height: 108,
    density: "mdpi" /* MDPI */,
};
const ANDROID_HDPI_ADAPTIVE_ICON = {
    foreground: 'android/icon/hdpi-foreground.png',
    platform: "android" /* ANDROID */,
    type: "adaptive-icon" /* ADAPTIVE_ICON */,
    background: 'android/icon/hdpi-background.png',
    format: "png" /* PNG */,
    width: 162,
    height: 162,
    density: "hdpi" /* HDPI */,
};
const ANDROID_XHDPI_ADAPTIVE_ICON = {
    platform: "android" /* ANDROID */,
    type: "adaptive-icon" /* ADAPTIVE_ICON */,
    foreground: 'android/icon/xhdpi-foreground.png',
    background: 'android/icon/xhdpi-background.png',
    format: "png" /* PNG */,
    width: 216,
    height: 216,
    density: "xhdpi" /* XHDPI */,
};
const ANDROID_XXHDPI_ADAPTIVE_ICON = {
    platform: "android" /* ANDROID */,
    type: "adaptive-icon" /* ADAPTIVE_ICON */,
    foreground: 'android/icon/xxhdpi-foreground.png',
    background: 'android/icon/xxhdpi-background.png',
    format: "png" /* PNG */,
    width: 324,
    height: 324,
    density: "xxhdpi" /* XXHDPI */,
};
const ANDROID_XXXHDPI_ADAPTIVE_ICON = {
    platform: "android" /* ANDROID */,
    type: "adaptive-icon" /* ADAPTIVE_ICON */,
    foreground: 'android/icon/xxxhdpi-foreground.png',
    background: 'android/icon/xxxhdpi-background.png',
    format: "png" /* PNG */,
    width: 432,
    height: 432,
    density: "xxxhdpi" /* XXXHDPI */,
};
exports.ANDROID_ADAPTIVE_ICON_RESOURCES = [
    ANDROID_LDPI_ADAPTIVE_ICON,
    ANDROID_MDPI_ADAPTIVE_ICON,
    ANDROID_HDPI_ADAPTIVE_ICON,
    ANDROID_XHDPI_ADAPTIVE_ICON,
    ANDROID_XXHDPI_ADAPTIVE_ICON,
    ANDROID_XXXHDPI_ADAPTIVE_ICON,
];
const ANDROID_LDPI_ICON = {
    platform: "android" /* ANDROID */,
    type: "icon" /* ICON */,
    src: 'android/icon/drawable-ldpi-icon.png',
    format: "png" /* PNG */,
    width: 36,
    height: 36,
    density: "ldpi" /* LDPI */,
};
const ANDROID_MDPI_ICON = {
    platform: "android" /* ANDROID */,
    type: "icon" /* ICON */,
    src: 'android/icon/drawable-mdpi-icon.png',
    format: "png" /* PNG */,
    width: 48,
    height: 48,
    density: "mdpi" /* MDPI */,
};
const ANDROID_HDPI_ICON = {
    platform: "android" /* ANDROID */,
    type: "icon" /* ICON */,
    src: 'android/icon/drawable-hdpi-icon.png',
    format: "png" /* PNG */,
    width: 72,
    height: 72,
    density: "hdpi" /* HDPI */,
};
const ANDROID_XHDPI_ICON = {
    platform: "android" /* ANDROID */,
    type: "icon" /* ICON */,
    src: 'android/icon/drawable-xhdpi-icon.png',
    format: "png" /* PNG */,
    width: 96,
    height: 96,
    density: "xhdpi" /* XHDPI */,
};
const ANDROID_XXHDPI_ICON = {
    platform: "android" /* ANDROID */,
    type: "icon" /* ICON */,
    src: 'android/icon/drawable-xxhdpi-icon.png',
    format: "png" /* PNG */,
    width: 144,
    height: 144,
    density: "xxhdpi" /* XXHDPI */,
};
const ANDROID_XXXHDPI_ICON = {
    platform: "android" /* ANDROID */,
    type: "icon" /* ICON */,
    src: 'android/icon/drawable-xxxhdpi-icon.png',
    format: "png" /* PNG */,
    width: 192,
    height: 192,
    density: "xxxhdpi" /* XXXHDPI */,
};
exports.ANDROID_ICON_RESOURCES = [
    ANDROID_LDPI_ICON,
    ANDROID_MDPI_ICON,
    ANDROID_HDPI_ICON,
    ANDROID_XHDPI_ICON,
    ANDROID_XXHDPI_ICON,
    ANDROID_XXXHDPI_ICON,
];
const ANDROID_LAND_LDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-land-ldpi-screen.png',
    format: "png" /* PNG */,
    width: 320,
    height: 240,
    density: "land-ldpi" /* LAND_LDPI */,
    orientation: "landscape" /* LANDSCAPE */,
};
const ANDROID_LAND_MDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-land-mdpi-screen.png',
    format: "png" /* PNG */,
    width: 480,
    height: 320,
    density: "land-mdpi" /* LAND_MDPI */,
    orientation: "landscape" /* LANDSCAPE */,
};
const ANDROID_LAND_HDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-land-hdpi-screen.png',
    format: "png" /* PNG */,
    width: 800,
    height: 480,
    density: "land-hdpi" /* LAND_HDPI */,
    orientation: "landscape" /* LANDSCAPE */,
};
const ANDROID_LAND_XHDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-land-xhdpi-screen.png',
    format: "png" /* PNG */,
    width: 1280,
    height: 720,
    density: "land-xhdpi" /* LAND_XHDPI */,
    orientation: "landscape" /* LANDSCAPE */,
};
const ANDROID_LAND_XXHDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-land-xxhdpi-screen.png',
    format: "png" /* PNG */,
    width: 1600,
    height: 960,
    density: "land-xxhdpi" /* LAND_XXHDPI */,
    orientation: "landscape" /* LANDSCAPE */,
};
const ANDROID_LAND_XXXHDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-land-xxxhdpi-screen.png',
    format: "png" /* PNG */,
    width: 1920,
    height: 1280,
    density: "land-xxxhdpi" /* LAND_XXXHDPI */,
    orientation: "landscape" /* LANDSCAPE */,
};
const ANDROID_PORT_LDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-port-ldpi-screen.png',
    format: "png" /* PNG */,
    width: 240,
    height: 320,
    density: "port-ldpi" /* PORT_LDPI */,
    orientation: "portrait" /* PORTRAIT */,
};
const ANDROID_PORT_MDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-port-mdpi-screen.png',
    format: "png" /* PNG */,
    width: 320,
    height: 480,
    density: "port-mdpi" /* PORT_MDPI */,
    orientation: "portrait" /* PORTRAIT */,
};
const ANDROID_PORT_HDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-port-hdpi-screen.png',
    format: "png" /* PNG */,
    width: 480,
    height: 800,
    density: "port-hdpi" /* PORT_HDPI */,
    orientation: "portrait" /* PORTRAIT */,
};
const ANDROID_PORT_XHDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-port-xhdpi-screen.png',
    format: "png" /* PNG */,
    width: 720,
    height: 1280,
    density: "port-xhdpi" /* PORT_XHDPI */,
    orientation: "portrait" /* PORTRAIT */,
};
const ANDROID_PORT_XXHDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-port-xxhdpi-screen.png',
    format: "png" /* PNG */,
    width: 960,
    height: 1600,
    density: "port-xxhdpi" /* PORT_XXHDPI */,
    orientation: "portrait" /* PORTRAIT */,
};
const ANDROID_PORT_XXXHDPI_SCREEN = {
    platform: "android" /* ANDROID */,
    type: "splash" /* SPLASH */,
    src: 'android/splash/drawable-port-xxxhdpi-screen.png',
    format: "png" /* PNG */,
    width: 1280,
    height: 1920,
    density: "port-xxxhdpi" /* PORT_XXXHDPI */,
    orientation: "portrait" /* PORTRAIT */,
};
exports.ANDROID_SPLASH_RESOURCES = [
    ANDROID_LAND_LDPI_SCREEN,
    ANDROID_LAND_MDPI_SCREEN,
    ANDROID_LAND_HDPI_SCREEN,
    ANDROID_LAND_XHDPI_SCREEN,
    ANDROID_LAND_XXHDPI_SCREEN,
    ANDROID_LAND_XXXHDPI_SCREEN,
    ANDROID_PORT_LDPI_SCREEN,
    ANDROID_PORT_MDPI_SCREEN,
    ANDROID_PORT_HDPI_SCREEN,
    ANDROID_PORT_XHDPI_SCREEN,
    ANDROID_PORT_XXHDPI_SCREEN,
    ANDROID_PORT_XXXHDPI_SCREEN,
];
/**
 * 20pt Icon
 *
 * - iPhone Notification (iOS 7+)
 * - iPad Notification (iOS 7+)
 */
const IOS_ICON_20_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-20.png',
    format: "png" /* PNG */,
    width: 20,
    height: 20,
};
const IOS_ICON_20_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-20@2x.png',
    format: "png" /* PNG */,
    width: 40,
    height: 40,
};
const IOS_ICON_20_PT_3X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-20@3x.png',
    format: "png" /* PNG */,
    width: 60,
    height: 60,
};
/**
 * 29pt Icon
 *
 * - iPhone Settings (iOS 7+)
 * - iPad Settings (iOS 7+)
 * - Apple Watch Companion Settings
 * - Apple Watch Notification Center
 */
const IOS_ICON_29_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-29.png',
    format: "png" /* PNG */,
    width: 29,
    height: 29,
};
const IOS_ICON_29_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-29@2x.png',
    format: "png" /* PNG */,
    width: 58,
    height: 58,
};
const IOS_ICON_29_PT_3X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-29@3x.png',
    format: "png" /* PNG */,
    width: 87,
    height: 87,
};
/**
 * 40pt Icon
 *
 * - iPhone Spotlight (iOS 7+)
 * - iPad Spotlight (iOS 7+)
 * - Apple Watch Home Screen
 */
const IOS_ICON_40_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-40.png',
    format: "png" /* PNG */,
    width: 40,
    height: 40,
};
const IOS_ICON_40_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-40@2x.png',
    format: "png" /* PNG */,
    width: 80,
    height: 80,
};
const IOS_ICON_40_PT_3X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-40@3x.png',
    format: "png" /* PNG */,
    width: 120,
    height: 120,
};
/**
 * 50pt Icon
 *
 * - iPad Spotlight (iOS 5,6)
 * - Apple Watch Home Screen
 */
const IOS_ICON_50_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-50.png',
    format: "png" /* PNG */,
    width: 50,
    height: 50,
};
const IOS_ICON_50_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-50@2x.png',
    format: "png" /* PNG */,
    width: 100,
    height: 100,
};
/**
 * 57pt Icon
 *
 * - iPhone App (iOS 5,6)
 */
const IOS_ICON_57_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon.png',
    format: "png" /* PNG */,
    width: 57,
    height: 57,
};
const IOS_ICON_57_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon@2x.png',
    format: "png" /* PNG */,
    width: 114,
    height: 114,
};
/**
 * 60pt Icon
 *
 * - iPhone App (iOS 7+)
 */
const IOS_ICON_60_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-60.png',
    format: "png" /* PNG */,
    width: 60,
    height: 60,
};
const IOS_ICON_60_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-60@2x.png',
    format: "png" /* PNG */,
    width: 120,
    height: 120,
};
const IOS_ICON_60_PT_3X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-60@3x.png',
    format: "png" /* PNG */,
    width: 180,
    height: 180,
};
/**
 * 72pt Icon
 *
 * - iPad App (iOS 5,6)
 */
const IOS_ICON_72_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-72.png',
    format: "png" /* PNG */,
    width: 72,
    height: 72,
};
const IOS_ICON_72_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-72@2x.png',
    format: "png" /* PNG */,
    width: 144,
    height: 144,
};
/**
 * 76pt Icon
 *
 * - iPad App (iOS 7+)
 */
const IOS_ICON_76_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-76.png',
    format: "png" /* PNG */,
    width: 76,
    height: 76,
};
const IOS_ICON_76_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-76@2x.png',
    format: "png" /* PNG */,
    width: 152,
    height: 152,
};
/**
 * 83.5pt Icon
 *
 * iPad Pro (12.9-inch)
 */
const IOS_ICON_83_5_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-83.5@2x.png',
    format: "png" /* PNG */,
    width: 167,
    height: 167,
};
/**
 * 1024px Icon
 *
 * - App Store
 */
const IOS_ICON_1024 = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-1024.png',
    format: "png" /* PNG */,
    width: 1024,
    height: 1024,
};
/**
 * 24pt Icon
 *
 * - Apple Watch Notification Center
 */
const IOS_ICON_24_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-24@2x.png',
    format: "png" /* PNG */,
    width: 48,
    height: 48,
};
/**
 * 27.5pt Icon
 *
 * - Apple Watch Notification Center
 */
const IOS_ICON_27_5_PT = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-27.5@2x.png',
    format: "png" /* PNG */,
    width: 55,
    height: 55,
};
/**
 * 44pt Icon
 *
 * - Apple Watch Home Screen
 */
const IOS_ICON_44_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-44@2x.png',
    format: "png" /* PNG */,
    width: 88,
    height: 88,
};
/**
 * 86pt Icon
 *
 * - Apple Watch Short Look
 */
const IOS_ICON_86_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-86@2x.png',
    format: "png" /* PNG */,
    width: 172,
    height: 172,
};
/**
 * 98pt Icon
 *
 * - Apple Watch Short Look
 */
const IOS_ICON_98_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-98@2x.png',
    format: "png" /* PNG */,
    width: 196,
    height: 196,
};
/**
 * 108pt Icon
 *
 * - Apple Watch Short Look
 */
const IOS_ICON_108_PT_2X = {
    platform: "ios" /* IOS */,
    type: "icon" /* ICON */,
    src: 'ios/icon/icon-108@2x.png',
    format: "png" /* PNG */,
    width: 216,
    height: 216,
};
exports.IOS_ICON_RESOURCES = [
    IOS_ICON_57_PT,
    IOS_ICON_57_PT_2X,
    IOS_ICON_20_PT,
    IOS_ICON_20_PT_2X,
    IOS_ICON_20_PT_3X,
    IOS_ICON_29_PT,
    IOS_ICON_29_PT_2X,
    IOS_ICON_29_PT_3X,
    IOS_ICON_24_PT,
    IOS_ICON_27_5_PT,
    IOS_ICON_44_PT_2X,
    IOS_ICON_86_PT_2X,
    IOS_ICON_98_PT_2X,
    IOS_ICON_108_PT_2X,
    IOS_ICON_40_PT,
    IOS_ICON_40_PT_2X,
    IOS_ICON_40_PT_3X,
    IOS_ICON_50_PT,
    IOS_ICON_50_PT_2X,
    IOS_ICON_60_PT,
    IOS_ICON_60_PT_2X,
    IOS_ICON_60_PT_3X,
    IOS_ICON_72_PT,
    IOS_ICON_72_PT_2X,
    IOS_ICON_76_PT,
    IOS_ICON_76_PT_2X,
    IOS_ICON_83_5_PT_2X,
    IOS_ICON_1024,
];
const IOS_SPLASH_568H_2X_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-568h@2x~iphone.png',
    format: "png" /* PNG */,
    width: 640,
    height: 1136,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_667H = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-667h.png',
    format: "png" /* PNG */,
    width: 750,
    height: 1334,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_2688H_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-2688h~iphone.png',
    format: "png" /* PNG */,
    width: 1242,
    height: 2688,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_2688H_LANDSCAPE_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape-2688h~iphone.png',
    format: "png" /* PNG */,
    width: 2688,
    height: 1242,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_1792H_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-1792h~iphone.png',
    format: "png" /* PNG */,
    width: 828,
    height: 1792,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_1792H_LANDSCAPE_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape-1792h~iphone.png',
    format: "png" /* PNG */,
    width: 1792,
    height: 828,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_2436H = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-2436h.png',
    format: "png" /* PNG */,
    width: 1125,
    height: 2436,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_2436H_LANDSCAPE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape-2436h.png',
    format: "png" /* PNG */,
    width: 2436,
    height: 1125,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_736H = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-736h.png',
    format: "png" /* PNG */,
    width: 1242,
    height: 2208,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_736H_LANDSCAPE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape-736h.png',
    format: "png" /* PNG */,
    width: 2208,
    height: 1242,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_LANDSCAPE_2X_IPAD = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape@2x~ipad.png',
    format: "png" /* PNG */,
    width: 2048,
    height: 1536,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_LANDSCAPE_IPADPRO = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape@~ipadpro.png',
    format: "png" /* PNG */,
    width: 2732,
    height: 2048,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_LANDSCAPE_IPAD = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Landscape~ipad.png',
    format: "png" /* PNG */,
    width: 1024,
    height: 768,
    orientation: "landscape" /* LANDSCAPE */,
};
const IOS_SPLASH_PORTRAIT_2X_IPAD = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Portrait@2x~ipad.png',
    format: "png" /* PNG */,
    width: 1536,
    height: 2048,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_PORTRAIT_IPADPRO = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Portrait@~ipadpro.png',
    format: "png" /* PNG */,
    width: 2048,
    height: 2732,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_PORTRAIT_IPAD = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default-Portrait~ipad.png',
    format: "png" /* PNG */,
    width: 768,
    height: 1024,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_2X_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default@2x~iphone.png',
    format: "png" /* PNG */,
    width: 640,
    height: 960,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_IPHONE = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default~iphone.png',
    format: "png" /* PNG */,
    width: 320,
    height: 480,
    orientation: "portrait" /* PORTRAIT */,
};
const IOS_SPLASH_2X_UNIVERSAL_ANYANY = {
    platform: "ios" /* IOS */,
    type: "splash" /* SPLASH */,
    src: 'ios/splash/Default@2x~universal~anyany.png',
    format: "png" /* PNG */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* PORTRAIT */,
};
exports.IOS_SPLASH_RESOURCES = [
    IOS_SPLASH_568H_2X_IPHONE,
    IOS_SPLASH_667H,
    IOS_SPLASH_2688H_IPHONE,
    IOS_SPLASH_2688H_LANDSCAPE_IPHONE,
    IOS_SPLASH_1792H_IPHONE,
    IOS_SPLASH_1792H_LANDSCAPE_IPHONE,
    IOS_SPLASH_2436H,
    IOS_SPLASH_2436H_LANDSCAPE,
    IOS_SPLASH_736H,
    IOS_SPLASH_736H_LANDSCAPE,
    IOS_SPLASH_LANDSCAPE_2X_IPAD,
    IOS_SPLASH_LANDSCAPE_IPADPRO,
    IOS_SPLASH_LANDSCAPE_IPAD,
    IOS_SPLASH_PORTRAIT_2X_IPAD,
    IOS_SPLASH_PORTRAIT_IPADPRO,
    IOS_SPLASH_PORTRAIT_IPAD,
    IOS_SPLASH_2X_IPHONE,
    IOS_SPLASH_IPHONE,
    IOS_SPLASH_2X_UNIVERSAL_ANYANY,
];
